### PR TITLE
Lock Frozen Ticks API

### LINKED ISSUE #6812: setFrozen or freezetick similar to firetick
### Is your feature request related to a problem?

Unlike with firetick, the freeze effect is only applied when inside powder snow for x ticks. Simply repeatedly setting the freezetick to max for a player doesnt solve it.

### Describe the solution you'd like.

Some way to specify that an entity is frozen, as if they were inside a powder snowblock with max freezetick.
There is already entity.isFrozen, but it only returns if it is frozen or not.

This would ofcourse not be removed when not in a powder snow block, and be a constant effect until it is "false"

### Describe alternatives you've considered.

Constantly setting the entities freezetick to max (doesnt work)
Sending a blockchange to the location of the entity

### Other

136946145-73542c0f-9f03-4a20-b234-ffed604c8142
When constantly setting freezeticks, this is the result.

### PR DESCRIPTION
**Pull request by Owen1212055**
Resolves https://github.com/PaperMC/Paper/issues/6812

Allows you to lock freeze ticks to a certain number, allowing you to prevent it from being updated nicely.

### CODE REVIEW

**Review by kashike — APPROVED** [review]
Not tested, but the code looks good.

File: patches/server/0839-Freeze-Override-API.patch [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
++                // Paper end - Frozen Override API
+ 
+                 attributemodifiable.addTransientModifier(new AttributeModifier(LivingEntity.SPEED_MODIFIER_POWDER_SNOW_UUID, "Powder snow slow", (double) f, AttributeModifier.Operation.ADDITION));
+             }
+@@ -4276,6 +4280,9 @@ public abstract class LivingEntity extends Entity {
+ 
+     @Override
+     public boolean canFreeze() {
++        if (this.frozenOverride == net.kyori.adventure.util.TriState.TRUE) {return true;} // Paper - Frozen Override API
++        else if (this.frozenOverride == net.kyori.adventure.util.TriState.FALSE) {return false;} // Paper - Frozen Override API
```
Comment (Machine-Maker) [inline patches/server/0839-Freeze-Override-API.patch]: I'm not sure about this. I think the `canFreeze` method should be left to its vanilla logic, but rather in places where canFreeze is used, check the override. Or maybe make another method that checks it and call that one in place of canFreeze

File: patches/server/0839-Freeze-Override-API.patch [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
++        } else {
++            this.entityData.set(Entity.DATA_TICKS_FROZEN, realFrozenTicks);
++        }
++    }
++    // Paper end - Frozen Override API
+ 
+     public void setTicksFrozen(int frozenTicks) {
+-        this.entityData.set(Entity.DATA_TICKS_FROZEN, frozenTicks);
++        updateFrozenEffect(); // Paper - Frozen Override API
++        this.realFrozenTicks = frozenTicks; // Paper - Frozen Override API
```
Comment (Owen1212055) [inline patches/server/0839-Freeze-Override-API.patch]: blocker, need to fix this. The field needs to be set first.

**Comment by Owen1212055** [comment]
Reworked pr to instead lock the frozen ticks, also made it persistent.

**Review by Machine-Maker — APPROVED** [review]

**Review by kennytv — APPROVED** [review]

**Pull request merged**